apps/sms/js/thread_list_ui.js

```diff
@@ -449,88 +455,80 @@ var ThreadListUI = {
   // please make sure url will also be revoked if new delete api remove threads
   // without calling removeThread in the future.
   delete: function thlui_delete(selected) {
-    function performDeletion() {
-    /* jshint validthis: true */
-
-      var threadIdsToDelete = [],
-          messageIdsToDelete = [],
-          threadCountToDelete = 0;
-
-      function exitEditMode() {
-        ThreadListUI.cancelEdit();
-        WaitingScreen.hide();
+    return Utils.confirm(
+      {
+        id: 'deleteThreads-confirmation-message',
+        args: { n: selected.length }
+      },
+      null,
+      {
+        text: 'delete',
+        className: 'danger'
       }
+    ).then(this.deleteThreadDraft.bind(this, selected));
+  },
 
-      function onAllThreadMessagesRetrieved() {
-        if (!--threadCountToDelete) {
-          MessageManager.deleteMessages(messageIdsToDelete);
-
-          threadIdsToDelete.forEach(function(threadId) {
-            ThreadListUI.deleteThread(threadId);
-          });
+  deleteThreadDraft: function thlui_deleteThreadDraft(selected) {
+    var threadIds = [], draftIds = [];
 
-          messageIdsToDelete = threadIdsToDelete = null;
+    WaitingScreen.show();
 
-          exitEditMode();
-        }
-      }
-
-      function onThreadMessageRetrieved(message) {
-        messageIdsToDelete.push(message.id);
-        return true;
+    selected.forEach((id) => {
+      // Coerce the threadId back to a number MobileMessageFilter and all
+      // other platform APIs expect this id to be a number.
+      var threadId = +id;
+      var isDraft = typeof Threads.get(threadId) === 'undefined';
+
+      if (isDraft) {
+        draftIds.push(threadId);
+        this.removeThread(threadId);
+      } else {
+        threadIds.push(threadId);
+        this.deleteThread(threadId);
       }
```

Comment:
wondering if we can call deleteThread for both drafts and threads... (I don't know if bug 1084298 changes this BTW)